tests/test_permissions.py

```diff
@@ -49,10 +60,24 @@ def test_has_perm_not_registered_for_site_permissions(self):
         ):
             self.assertIsNone(
                 SitePermissionBackend().has_perm(
-                    usersite.user, "polls.change_poll", poll
+                    usersite.user, self.change_permission, obj
                 )
             )
 
     def test_has_perm_no_obj_passed(self):
         user = UserFactory()
         self.assertIsNone(SitePermissionBackend().has_perm(user, "polls.change_poll"))
+
+
+class FKPermissionsTestCase(BasePermissionsTests, TestCase):
+    change_permission = "polls.change_poll"
+
+    def factory(self, site):
+        return PollFactory(site=site)
+
+
+class M2MRulesTestCase(BasePermissionsTests, TestCase):
+    change_permission = "restaurants.change_restaurant"
+
+    def factory(self, site):
+        return RestaurantFactory(sites=[site])
```

Comment:
I think you should add three more tests on this test case - `test_has_perm` for a case with two sites where the user has perms for both, `test_has_perm` for a case with two sites where the user has perms for one and `test_has_perm_no_site_access` for a case with two sites where the user doesn't have access to either. If you change the factory method to take `*sites` instead of `site` this should be easy enough.